addons/website_slides/tests/test_slide_channel.py

```diff
@@ -274,6 +275,46 @@ def _get_completion_time_pdf(*args, **kwargs):
         self.assertEqual(13.37, round(slides_1.completion_time, 2))
         self.assertEqual(123.0, slides_2.completion_time)
 
+    def test_if_unlink_and_unpublish_channel_slides_sends_competition_email(self):
+        """Check that course participants do not receive a "course completed" email when slides are deleted."""
+        def were_emails_sent():
+            new_mails = self._new_mails.filtered(lambda m: m.model == 'slide.channel.partner')
+            return len(new_mails) > 0
+
+        # 0. Setup: make sure there's channel with slides, that nobody completed
+        self.assertGreater(len(self.channel.slide_ids), 1, "Channel shall have at least two slides.")
+        self.assertGreater(len(self.channel.channel_partner_ids), 0, "Channel shall have at least one participant (channel_partner_ids)")
+        self.assertEqual(self.channel.members_completed_count, 0, "Someone received competition mail notification (nobody should have)")
+
+        # 1. Action: Unpublish one slide
+        slide_published = self.channel.slide_ids.filtered('is_published')[:1]
+        self.assertEqual(len(slide_published), 1, "There are no published slides, can't perform 'Action: Unpublish one slide'")
+        with self.mock_mail_gateway():
+            slide_published.with_user(self.user_manager).is_published = False
+        self.assertFalse(were_emails_sent(), "Unpublishing single slide sends competition mail to it's users. It shouldn't!")
+
+        # 2. Action: Unpublish all slides
+        slide_published_cnt = len(self.channel.slide_ids.filtered('is_published'))
+        self.assertGreater(slide_published_cnt, 0, "There are no published slides, can't perform 'Action: Unpublish all slides'")
+        with self.mock_mail_gateway():
+            self.channel.slide_ids.with_user(self.user_manager).is_published = False
+        self.assertFalse(were_emails_sent(), "Unpublishing all slides slide sends competition mail to it's users. It shouldn't!")
+
+        # 3. Action: Delete one unpublished slide
+        slide = self.channel.slide_ids.filtered(lambda s: not s.is_published)[:1]
+        slide.question_ids.with_user(self.user_manager).unlink()  # Slide questions (if any) needs to be deleted in order to delete the slide
+        with self.mock_mail_gateway():
+            slide.with_user(self.user_manager).unlink()
+        self.assertFalse(were_emails_sent(), "Deleting single side sends competition mail to it's users. It shouldn't!")
+
+        # 4.Action: Delete all unpublished slides
+        slides = self.channel.slide_ids.filtered(lambda s: not s.is_published)
+        slides.question_ids.with_user(self.user_manager).unlink()  # Slide questions (if any) needs to be deleted in order to delete the slide
```

Comment:
You are controlling the test environment. Are there any questions in the test data?

If yes,
```suggestion
        slides.question_ids.with_user(self.user_manager).unlink()  # Slide questions need to be deleted in order to delete the slide
```
If not:

```suggestion
```
If someone modifies the test data to add questions when there are none and it breaks this test, it will be their job to update this test.

(same above of course).